src/commands/gemini.py

```diff
@@ -0,0 +1,370 @@
+
+            err = is_valid_ext_size(author, attachment)
+            if err:
+                errors.append(err)
+            if err is None:
+                file_ref, err = await upload_or_return_file_ref(attachment)
+                if file_ref:
+                    attachment_ref = file_ref
+                    response_image_url = attachment.url
+                if err:
+                    logging.error(
+                        f"GEMINI: {author} encountered an error uploading an attachment to Gemini: {ERROR_MESSAGES[err]}"
+                    )
+                    errors.append(err)
+
+        # If only attachment was provided but did not upload successfully
+        if message is None and len(errors) != 0:
+            logging.error(
+                f"GEMINI: {author} passed an empty query and uploaded an invalid attachment to Gemini."
+            )
+            return [
+                Embed(
+                    title="Ask Gemini",
+                    description="The attachment is invalid and no text input is provided.",
+                    color=LIGHT_YELLOW,
+                )
+            ]
+
+        response_embed, err = await self.prompt_gemini(
+            author=author, input_msg=message, attachment=attachment_ref
+        )
+
+        # If response is none, it means something went wrong, so directly go to the error embed
+        if response_embed is not None:
+
+            # Only set the response_embed image if image is provided and
+            if response_image_url is not None:
+                response_embed.set_image(url=response_image_url)
+
+            response_embeds.append(response_embed)
+
+        # The last encountered error is used for the embed
+        if err is not None:
+            errors.append(err)
+
+        if len(errors) != 0:
+            err_embed = get_error_embed(errors)
+            response_embeds.append(err_embed)
+
+        # The chat instance maintains a chat convo by sending all previous messages as input every time
+        # This can easily exhaust the free tier of Gemini API, so choosing to clear the history every 50 messages
+        if len(self.chat.history) >= 50:
+            self.chat.history([])
+            delete_files()
+
+        return response_embeds
+
+
+async def upload_or_return_file_ref(attachment) -> (File, Errors):
+    """Uploads the image to the Google Gemini Project
+    Stored for 48 hours by default"""
+
+    file_name = os.path.splitext(attachment.filename)[0]
+
+    try:
+        # If image is already uploaded to
+        file_ref = return_genai_file_ref(file_name)
+
+        if file_ref:
+            return file_ref, None
+
+        # If new image
+        else:
+            with tempfile.TemporaryDirectory() as temp:
+
+                await attachment.save(f"{temp}/{attachment.filename}")
+
+                file_ref = genai.upload_file(
+                    path=f"{temp}/{attachment.filename}", display_name=file_name
```

Comment:
I tested this and it just uses the first uploaded image with the same name, so you'll need to add some logic to rename the file if it already exists or delete the old file.

Reply:
I reckon there should be some sort of hash or randomly generated ID that gets prepended or appended to the file name

Reply:
Using a hash as the filename itself is a good idea because we can then avoid people uploading the same media files using different file names. That does not seem very common though. For now, I'll prepend the hash to the filename.